test/chain-blockstore-test.js

```diff
@@ -0,0 +1,279 @@
+        db.start = dbStart;
+      });
+
+      it('should reopen the chain', async () => {
+        await chain.close();
+        await chain.open();
+      });
+
+      it('should handle blockstore failing to prune', async () => {
+        const db = chain.db;
+
+        // backup start method
+        const dbStart = db.start;
+
+        db.start = function() {
+          dbStart.call(this);
+
+          this.blocksBatch.commitPrunes = function() {
+            throw new Error('Failed to prune.');
+          };
+        };
+
+        let err;
+
+        try {
+          await chain.reset(1);
+        } catch (e) {
+          err = e;
+        }
+
+        assert(err);
+        assert(err.message, 'Failed to prune.');
+
+        // recover start method.
+        db.start = dbStart;
+      });
+
+      it('should reopen the chain', async () => {
+        await chain.close();
+        await chain.open();
+      });
+    });
+  }
+
+  describe('Chain File BlockStore Integration 2', function() {
+    const location = testdir('chain-blockstore-2');
+    const workers = new WorkerPool({
+      enabled: true,
+      size: 2
+    });
+
+    const MINE_BLOCKS = 10;
+
+    let blocks, chain, miner, cpu, wallet;
+
+    before(async () => {
+      await rimraf(location);
+
+      blocks = await blockStoreTypes['File'](location);
+
+      chain = new Chain({
+        prefix: location,
+        blocks,
+        network,
+        workers
+      });
+
+      miner = new Miner({ chain, workers });
+      cpu = miner.cpu;
+      wallet = new MemWallet({ network });
+
+      await blocks.open();
+      await chain.open();
+      await miner.open();
+    });
+
+    after(async () => {
+      await miner.close();
+      await chain.close();
+      await blocks.close();
+
+      await rimraf(location);
+    });
+
+    it('should add addrs to miner', async () => {
+      miner.addresses.length = 0;
+      miner.addAddress(wallet.getReceive());
+    });
+
+    it(`should mine ${MINE_BLOCKS} blocks`, async () => {
+      for (let i = 0; i < MINE_BLOCKS; i++) {
+        const block = await cpu.mineBlock();
+        assert(block);
+        assert(await chain.add(block));
+      }
+    });
+
+    it('should handle file blockstore failing to prune', async () => {
+      const db = chain.db;
+
+      // backup start method
+      const dbStart = db.start;
+
+      db.start = function() {
+        dbStart.call(this);
+
+        this.blocksBatch.commitPrunes = function() {
+          if (this.prunes.length === 0) {
+            this.committedPrunes = true;
+            return;
+          }
+
+          throw new Error('Failed to prune.');
+        };
+      };
+
+      let err;
+
+      assert.strictEqual(chain.tip.height, MINE_BLOCKS);
+      try {
+        await chain.reset(1);
+      } catch (e) {
+        err = e;
+      }
+
+      assert(err);
+      assert(err.message, 'Failed to prune.');
+
+      // recover start method.
+      db.start = dbStart;
```

Comment:
should this move up before the assertions? if this test fails and this restore isnt executed, wont it have downstream effects on all following tests?